graylog2-server/src/test/java/org/graylog/plugins/views/search/rest/FieldTypesResourceTest.java

```diff
@@ -138,28 +119,22 @@ public void byStreamChecksPermissionsForStream() {
                 .allowStream("4242")
                 .build();
 
+        final FieldTypesForStreamsRequest req = FieldTypesForStreamsRequest.Builder.builder()
+                .streams(ImmutableSet.of("2323", "4242"))
+                .build();
 
-        final MappedFieldTypesService dependency = Mockito.mock(MappedFieldTypesService.class);
-        Mockito.when(dependency.fieldTypesByStreamIds(Mockito.any(), Mockito.any())).thenAnswer(invocation -> {
-            final Collection<String> streamIDs = invocation.getArgument(0);
+        final MappedFieldTypesService fieldTypesService = (streamIds, timeRange) -> {
             // for each streamID return a field that's called exactly like the streamID
-            return streamIDs.stream().map(streamID -> MappedFieldTypeDTO.create(streamID, FieldTypes.Type.builder().type("text").build())).collect(Collectors.toSet());
-        });
-
-        final FieldTypesResource resource = new FieldTypesResource(dependency);
-
-        final Set<MappedFieldTypeDTO> fields = resource.byStreams(
-                FieldTypesForStreamsRequest.Builder.builder()
-                        .streams(ImmutableSet.of("2323", "4242"))
-                        .build(),
-                searchUser
-        );
-
-        // field for allowed stream has to be present
-        assertThat(fields.stream().anyMatch(f -> f.name().equals("2323"))).isTrue();
-        assertThat(fields.stream().anyMatch(f -> f.name().equals("4242"))).isTrue();
+            return streamIds.stream().map(streamID -> MappedFieldTypeDTO.create(streamID, FieldTypes.Type.builder().type("text").build())).collect(Collectors.toSet());
```

Comment:
It is up to you, but I would move "FieldTypes.Type.builder().type("text").build()" to a separaten, named variable, just to increase the readability of the lamda.